### PR TITLE
Migrate code excerpter packages to null safety

### PR DESCRIPTION
**Pull request by parlough**
Updates the code excerpter packages to the null safety. Cleans up some code with new language features and standards along the way.

Fixes https://github.com/dart-lang/site-shared/issues/143

For review: @khanhnwin

### CODE REVIEW

**Comment by parlough** [comment]
@RedBrogdon Can you review this or assign someone on the devrel team?

@kevmoo Could you also take a look if you get a chance?

Thanks!

File: packages/code_excerpt_updater/lib/code_excerpt_updater_cli.dart [hunk 1]
```diff
@@ -9,11 +9,10 @@ import 'package:logging/logging.dart';
 
 import 'src/code_excerpt_updater.dart';
 import 'src/logger.dart';
-import 'src/nullable.dart';
 
-const _commandName = 'code_excerpt_updater';
```
Comment (kevmoo) [inline packages/code_excerpt_updater/lib/code_excerpt_updater_cli.dart:14]: Do we need to add types here? Adds a lot of noise IMHO.

**Review by kevmoo — APPROVED** [review]
in general, LGTM – careful about changes not related to the PR intent.

File: packages/code_excerpt_updater/lib/code_excerpt_updater_cli.dart (hunk 1 above)
Comment (parlough) [reply packages/code_excerpt_updater/lib/code_excerpt_updater_cli.dart:14]: I guess I somehow got used to adding types to top-level variables. I'll remove these changes as they're not related to the intention.

**Comment by khanhnwin** [comment]
Hey @parlough! Sorry I missed this PR. I'll give it a review by EOD tomorrow!

File: packages/analysis_options.yaml [hunk 2]
```diff
@@ -10,7 +17,7 @@ linter:
     - avoid_catching_errors
```
Comment (domesticmouse) [inline packages/analysis_options.yaml:17]: Can we migrate from handpicked rules to the `lints` package?

**Review by domesticmouse — APPROVED** [review]

File: packages/analysis_options.yaml (hunk 2 above)
Comment (parlough) [reply packages/analysis_options.yaml:17]: I can follow up with this and some other analysis related cleanup afterwards.

**Review by johnpryan — APPROVED** [review]

**Comment by parlough** [comment]
Can someone squash and merge this? I don't have commit access to site-shared. Thanks!

**Pull request merged**